src/api.hpp

```diff
@@ -154,6 +186,28 @@ EXTERN ComputeStatus one_off(const char* biom_filename, const char* tree_filenam
                              const char* unifrac_method, bool variance_adjust, double alpha,
                              bool bypass_tips, unsigned int threads, mat_t** result);
 
+
+/* Compute UniFrac - condensed form, in-memory
+ *
+ * table <biom> a constructed BIOM object
+ * tree <BPTree> a constructed BPTree object
+ * unifrac_method <const char*> the requested unifrac method.
+ * variance_adjust <bool> whether to apply variance adjustment.
+ * alpha <double> GUniFrac alpha, only relevant if method == generalized.
+ * bypass_tips <bool> disregard tips, reduces compute by about 50%
+ * threads <uint> the number of threads to use.
+ * result <mat_t**> the resulting distance matrix in condensed form, this is initialized within the method so using **
+ *
+ * one_off_inmem returns the following error codes:
+ *
+ * okay           : no problems encountered
+ * unknown_method : the requested method is unknown.
+ * table_empty    : the table does not have any entries
+ */
+EXTERN ComputeStatus one_off_inmem(support_biom_t *table_data, support_bptree_t *tree_data,
```

Comment:
@sfiligoi does anything unusual stand out here? It looks like the compiler is not honoring the `extern "C"`:

```bash
$ nm -D libssu.so | grep _inmem
00000000000ca9f0 T _Z13one_off_inmemPK12support_biomPK14support_bptreePKcbdbjPP3mat
00000000000c8a20 T _Z17one_off_inmem_cppRN2su4biomERNS_6BPTreeEPKcbdbjPP3mat
```

As far as I know, both `support_biom_t` and `support_bptree_t` are entirely native C types. 

It's not that `extern` isn't working as some functions are extern'd unmangled:

```bash
$ nm -D libssu.so | grep "one_off"
00000000000ca180 T faith_pd_one_off
00000000000cab20 T one_off
00000000000cc270 T one_off_matrix
00000000000cd310 T one_off_matrix_fp32
00000000000ca9f0 T _Z13one_off_inmemPK12support_biomPK14support_bptreePKcbdbjPP3mat
00000000000cb1e0 W _Z16one_off_matrix_TId13mat_full_fp64E14compute_statusPKcS3_S3_bdbjS3_PPT0_
00000000000cc280 W _Z16one_off_matrix_TIf13mat_full_fp32E14compute_statusPKcS3_S3_bdbjS3_PPT0_
00000000000c8a20 T _Z17one_off_inmem_cppRN2su4biomERNS_6BPTreeEPKcbdbjPP3mat
```

I'm going to continue to tinker, but wanted to ping incase I was missing something obvious

Reply:
Found the issue